transactor/src/test/scala/io/mediachain/transactor/JournalClusterSpec.scala

```diff
@@ -0,0 +1,172 @@
+package io.mediachain.transactor
+
+import org.specs2.specification.{AfterAll, BeforeAll}
+
+import java.util.concurrent.{BlockingQueue, LinkedBlockingQueue, TimeUnit}
+import scala.concurrent.{Future, Await}
+import scala.concurrent.duration.Duration
+import scala.concurrent.ExecutionContext.Implicits.global
+import cats.data.Xor
+
+import Types._
+
+object JournalClusterSpec extends io.mediachain.BaseSpec
+  with BeforeAll
+  with AfterAll
+{
+  val timeout = Duration(10, TimeUnit.SECONDS)
+
+  def is =
+    sequential ^
+  s2"""
+  JournalStateMachine is consistent in a cluster:
+   - the cluster can insert and update the journal $insertAndUpdateJournal
+   - all clients see the same chain mappings $checkChainViews
+   - all clients see the same current block $checkCurrentBlockViews
+   - all clients received the same journal commit events $checkCommitEvents
+   - the cluster can generate some blocks $generateBlocks
+   - all clients received the same journal block events $checkBlockEvents
+  """
+
+  def beforeAll() {
+    JournalClusterSpecContext.setup()
+  }
+  
+  def afterAll() {
+    JournalClusterSpecContext.shutdown()
+  }
+  
+  def insertAndUpdateJournal = {
+    val context = JournalClusterSpecContext.context
+    val ops = context.cluster.dummies.map(insertAndUpdate(_))
+    val res = ops.map(Await.result(_, timeout))
+    context.refs = res.map {
+      case Xor.Left(_) => null
+      case Xor.Right(tuple) => tuple
+    }
+    
+    (res(0) must beRightXor) and
+    (res(1) must beRightXor) and
+    (res(2) must beRightXor)
+  }
+  
+  private def insertAndUpdate(dummy: DummyContext)
+  : Future[Xor[JournalError, (Reference, Reference)]] = {
```

Comment:
Yeah, we should replace these `Future[Xor...` with a `XorF` to make the nesting less insane http://eed3si9n.com/herding-cats/stacking-future-and-either.html#XorT+datatype